examples/ColumnGroupsResizeReorderExample.js

```diff
@@ -0,0 +1,196 @@
+    for (var groupKey in this.state.columnOrder) {
+      for (const cKey of this.state.columnOrder[groupKey]) {
+        if (cKey == columnKey) {
+          return groupKey;
+        }
+      }
+    }
+    return null;
+  }
+
+  _onColumnReorderEndCallback = (event) => {
+    var columnGroup = this.findColumnGroup(event.reorderColumn);
+    var columnOrder = this.state.columnOrder[columnGroup].filter((columnKey) => {
+      return columnKey !== event.reorderColumn;
+    });
+
+    var columnBeforeIndex = columnOrder.indexOf(event.columnBefore);
+    var columnAfterIndex = columnOrder.indexOf(event.columnAfter);
+
+    if(columnBeforeIndex == -1 && columnAfterIndex == -1) {
+      return;
+    } else if (columnAfterIndex != -1) {
+      columnOrder.splice(columnAfterIndex, 0, event.reorderColumn);
+    } else {
+      if (fixedColumns.indexOf(event.reorderColumn) !== -1) {
+        columnOrder.splice(fixedColumns.length - 1, 0, event.reorderColumn)
+      } else {
+        columnOrder.push(event.reorderColumn);
+      }
+    }
+    var newColumnOrder = {...this.state.columnOrder};
+    newColumnOrder[columnGroup] = columnOrder;
+    this.setState({
+      columnOrder: newColumnOrder,
+    });
+  }
+
+  _onColumnGroupResizeEndCallback = (newColumnGroupWidth, columnGroupKey) => {
+    var columnWidths = {...this.state.columnWidths};
+    var currentColumns = this.state.columnOrder[columnGroupKey];
+    var totalWidth = 0;
+    for (var key of currentColumns) {
+      totalWidth += columnWidths[key]  
+    }
+
+    const widthChange = newColumnGroupWidth - totalWidth;
+    const perColumnChange = Math.floor(widthChange / currentColumns.length)
+    const remainderWidth = Math.abs(widthChange) % currentColumns.length;
+
+    for (var key of currentColumns) {
+      columnWidths[key] += perColumnChange;
+    }
+    columnWidths[Object.keys(currentColumns)[0]] += remainderWidth;
+    
+    this.setState({
+      columnWidths,
+    });
+  }
+
+  _onColumnResizeEndCallback = (newColumnWidth, columnKey) => {
+    this.setState(({columnWidths}) => ({
+      columnWidths: {
+        ...columnWidths,
+        [columnKey]: newColumnWidth,
+      }
+    }));
+  }
+
+  render() {
+    var { dataList, columnWidths, columnOrder } = this.state;
+    var onColumnGroupReorderEndCallback= this._onColumnGroupReorderEndCallback;
+    var onColumnReorderEndCallback= this._onColumnReorderEndCallback;
+    var onColumnGroupResizeEndCallback= this._onColumnGroupResizeEndCallback;
+    var onColumnResizeEndCallback= this._onColumnResizeEndCallback;
+    
+    return (
+      <Table
+        rowHeight={30}
+        groupHeaderHeight={30}
+        headerHeight={30}
+        rowsCount={dataList.getSize()}
+        width={1000}
+        height={500}
+        {...this.props}>
+        {this.state.columnGroupOrder.map(function (columnGroupKey, i) {
+          return <ColumnGroup
+            key={i}
+            header={
+              <Plugins.ResizeReorderCell
```

Comment:
Do ResizeReorder have to be combined in a single component or can they be split apart into a 'resize' and a 'reorder' if that makes them easier to use and more focused?

Reply:
We should be able to make them HOC's or use React hooks to implement them as more cross-cutting concerns.

Reply:
Do you mean, we will create 2 plugins ResizeHandle and ReorderHandle and it will be user's reponsibility to create a HOC to combine them ?

Reply:
Right.